src/github.com/matrix-org/dendrite/roomserver/input/events_test.go

```diff
@@ -0,0 +1,108 @@
+package input
+
+import (
+	"github.com/matrix-org/dendrite/roomserver/types"
+	"testing"
+)
+
+func benchmarkStateEntryMapLookup(entries, lookups int64, b *testing.B) {
+	var list []types.StateEntry
+	for i := int64(0); i < entries; i++ {
+		list = append(list, types.StateEntry{types.StateKeyTuple{i, i}, i})
+	}
+
+	for i := 0; i < b.N; i++ {
+		entryMap := stateEntryMap(list)
+		for j := int64(0); j < lookups; j++ {
+			entryMap.lookup(types.StateKeyTuple{j, j})
+		}
+	}
+}
+
+func BenchmarkStateEntryMap100Lookup10(b *testing.B) {
+	benchmarkStateEntryMapLookup(100, 10, b)
+}
+
+func BenchmarkStateEntryMap1000Lookup100(b *testing.B) {
+	benchmarkStateEntryMapLookup(1000, 100, b)
+}
+
+func BenchmarkStateEntryMap100Lookup100(b *testing.B) {
+	benchmarkStateEntryMapLookup(100, 100, b)
+}
+
+func BenchmarkStateEntryMap1000Lookup10000(b *testing.B) {
+	benchmarkStateEntryMapLookup(1000, 10000, b)
+}
+
+func TestStateEntryMap(t *testing.T) {
+	entryMap := stateEntryMap([]types.StateEntry{
+		{types.StateKeyTuple{1, 1}, 1},
+		{types.StateKeyTuple{1, 3}, 2},
+		{types.StateKeyTuple{2, 1}, 3},
+	})
+
+	testCases := []struct {
+		inputTypeNID  int64
+		inputStateKey int64
+		wantOK        bool
+		wantEventNID  int64
+	}{
+		{0, 0, false, 0},
+		{1, 1, true, 1},
+		{1, 3, true, 2},
+		{2, 1, true, 3},
+		{1, 2, false, 0},
+		{3, 1, false, 0},
```

Comment:
Please comment why the values of `wantOK` and `wantEventNID` are the values mentioned here like you did on previous tests you've done in Go. I have no idea why `3,1` is not valid for example...